sklearn/utils/__init__.py

```diff
@@ -402,37 +423,41 @@ def _safe_assign(X, values, *, row_indexer=None, column_indexer=None):
         X[row_indexer, column_indexer] = values
 
 
+def _get_column_indices_for_bool_or_int(key, n_columns):
+    # Convert key into positive indexes
+    try:
+        idx = _safe_indexing(np.arange(n_columns), key)
+    except IndexError as e:
+        raise ValueError(
+            "all features must be in [0, {}] or [-{}, 0]".format(
```

Comment:
Nitpick: Once ruff told me that raise Error statements should only contain strings to avoid raising another error, e.g. in a wrong f-string.